src/otx/algorithms/visual_prompting/tasks/inference.py

```diff
@@ -191,29 +186,31 @@ def get_model(config: DictConfig, state_dict: Optional[OrderedDict] = None):
                 "No trained model in project yet. Created new model with '%s'",
                 self.model_name,
             )
-        elif ("path" in otx_model.model_adapters) and (
-            otx_model.model_adapters.get("path").endswith(".ckpt")  # type: ignore[attr-defined]
-        ):
-            # pytorch lightning checkpoint
-            if not otx_model.model_adapters.get("resume"):
-                # If not resuming, just load weights in LightningModule
-                logger.info("Load pytorch lightning checkpoint.")
+        elif otx_model.model_adapters.get("resume", False):
+            # If resuming, pass this part to load checkpoint in Trainer
+            logger.info(f"To resume {otx_model.model_adapters.get('path')}, the checkpoint will be loaded in Trainer.")
+
         else:
-            # pytorch checkpoint saved by otx
+            # Load state_dict
             buffer = io.BytesIO(otx_model.get_data("weights.pth"))
             model_data = torch.load(buffer, map_location=torch.device("cpu"))
-            if model_data.get("model", None) and model_data.get("config", None):
+            if model_data.get("state_dict", None) and model_data.get("pytorch-lightning_version", None):
+                # Load state_dict from pytorch lightning checkpoint or weights.pth saved by visual prompting task
+                state_dict = model_data["state_dict"]
+
+            elif model_data.get("model", None) and model_data.get("config", None):
+                # Load state_dict from checkpoint saved by otx other tasks
```

Comment:
Is it possible to load a checkpoint from other tasks in visual prompting?

Reply:
Loading weights is possible, but resuming is not.
I planned that `weights.pth` saved by visual prompting can be used for resuming, not other tasks.
Do you think it is required?

Reply:
Ah actually I thought that it's impossible to load other tasks' model weight in visual prompting. Could you explain one example for that?

Reply:
Currently, we cannot test it because there is no task supporting ViT-B as backbone AFAIK.
But, for example, segmentation task saves `weights.pth` with metas, `model`, `config`, `labels`, and `VERSION`.
I think if we have ViT-B weights trained on other tasks, we can use its weights as pretrained weights.
![image](https://github.com/openvinotoolkit/training_extensions/assets/97221135/00b2678e-dd14-4a61-9e25-6ac82485979a)

Reply:
If you think it is unnecessary function for now, it would be better to remove this part.

Reply:
I think it's better to leave it as it is. Thanks for explanation.